haiku-apps/xrs/xrs-1.2.recipe

```diff
@@ -0,0 +1,49 @@
+SUMMARY="A simple rhythm station for creating loops"
+DESCRIPTION="XRS is a useful rhythm station for creating your own \
+looping music and could be used to create repeating background music \
+for videos or even a ringtone for your phone."
+HOMEPAGE="https://github.com/HaikuArchives/XRS"
+COPYRIGHT="2001-2003 Giuseppe Gargaro & Andrea Anzani"
+LICENSE="MIT"
+REVISION="1"
+srcGitRev="d91e2b8b808ace64b3032a75728b9fa3c43760f0"
+SOURCE_URI="https://github.com/HaikuArchives/xrs/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="11c656e544e59f345296f03f7c9ba513365f699cdb51608591a9e9e8a9dcbe6b"
+SOURCE_FILENAME="XRS-$portVersion-$srcGitRev.tar.gz"
+SOURCE_DIR="XRS-$srcGitRev"
+PATCHES="xrs-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 x86_64"
+
+PROVIDES="
+	xrs = $portVersion
+	app:XRS = $portVersion
+	"
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+BUILD_PREREQUIRES="
+	makefile_engine
+	cmd:g++
+	cmd:make
+	"
+
+BUILD()
+{
+	cd XrsPalm
+	cd src
+	make OBJ_DIR=objects
+}
+
+INSTALL()
+{
+	mkdir -p $appsDir
+	cd XrsPalm
+	cp XRS $appsDir
+	cp -r Skin $appsDir
```

Comment:
The app builds fine on x86_gcc2. (I did not try on x86_64 yet.)
The About box is not perfect, but that's another story.

Anyway, here, we have a small issue about the layout. This is what we have:
* /system/apps/XRS (the runtime)
* /system/apps/Skin/

We should use one of these alternate layouts:
* Solution 1
  * /system/apps/XRS/ (a subdirectory)
  * /system/apps/XRS/XRS (the runtime)
  * /system/apps/XRS/Skin/

or
* Solution 2
  * /system/apps/XRS (the runtime)
  * /system/data/XRS/ (a subdirectory)
  * /system/data/XRS/Skin/

The layout in solution 1 is similar to what we have in #3354 for BePhoneBook.
Solution 2 would be better imo, but this would require some changes in the app.

BTW, if you try solution 2 for XRS, would you like to try it for BePhoneBook too?
Note: it's not easy. I tried for BePhoneBook but did not succeed. Also, most contributors are OK with solution 1, which is much easier as it does not require changing the sources.

Reply:
@fbrosson I'll try both and see what I can do 😃 
As for testing on x86_64, I built this on x86_64, so it should build.

Reply:
@fbrosson I can confirm that method 2 is hard to do... the name of the binary and the source directory is the same, and haikuporter keeps telling me that I can't overwrite directory with nondirectory :(
On the other hand, I have method 2 running! Will push the commit up in a moment 😄

Reply:
Yeah, I tried this for method 1:
```diff
diff --git a/Source/windows.cpp b/Source/windows.cpp
index 826737b..5afa079 100644
--- a/Source/windows.cpp
+++ b/Source/windows.cpp
@@ -1,3 +1,8 @@
+#ifdef __HAIKU__
+#include <FindDirectory.h>
+#include <StorageDefs.h>
+#endif
```
```diff
 void MainWindow::MessageReceived(BMessage* msg)
 {
+	char readmePath[B_PATH_NAME_LENGTH];
-	BString Temp2(BePhonePath.Path());
 	BRoster MyRoster;
```
```diff
@@ -470,11 +476,11 @@ void MainWindow::MessageReceived(BMessage* msg)
 		break;
 	
 		case 'help':
-		Temp2+="/Doc/Readme.html";
-		entry.SetTo(Temp2.String());
-		entry.GetRef(&ref);
-		MyRoster.Launch(&ref,0,(char **)NULL);
-		
+		if (find_directory(B_SYSTEM_DATA_DIRECTORY, -1, false, readmePath, sizeof(readmePath)) == B_OK) {
+			entry.SetTo(strcpy(readmePath+strlen(readmePath),"/BePhoneBook/Doc/Readme.html"));
+			entry.GetRef(&ref);
+			MyRoster.Launch(&ref,0,(char **)NULL);
+		}
 		break;
...
but it did not work. So I made a mix between solution 1 and 2, to see if I could force the app to read the html file from the `$dataDir` without having to patch the code that looks for the Skin files. Of course `INSTALL` needs a small change:
```diff
 INSTALL()
 {
-	mkdir -p $appsDir/BePhoneBook
+	mkdir -p $appsDir/BePhoneBook $dataDir/BePhoneBook
 	cp objects/BePhoneBook $appsDir/BePhoneBook
 	cp -r Skins $appsDir/BePhoneBook
-	cp -r Doc $appsDir/BePhoneBook
+	cp -r Doc $dataDir/BePhoneBook
 	addAppDeskbarSymlink $appsDir/BePhoneBook
 }
```
The "mix" did not work, so decided it was not worth the trouble.
Just posting this here in case someone finds what is wrong in my attempt to use find_directory.

Feel free to choose method 2!

Reply:
[`BPathFinder`](https://www.haiku-os.org/docs/api/classBPathFinder.html) is more suited for this use case than `find_directory`.